packages/agent/src/agent-main.ts

```diff
@@ -46,10 +47,32 @@ export async function agentMain(argv: string[]): Promise<App> {
   const { baseUrl, clientId, clientSecret, agentId } = args;
 
   const medplum = new MedplumClient({ baseUrl, clientId });
-  await medplum.startClientLogin(clientId, clientSecret);
+
+  let loggedIn = false;
+  while (!loggedIn) {
+    try {
+      await medplum.startClientLogin(clientId, clientSecret);
```

Comment:
Hm, this might be too broad.  For example, if the client credentials are wrong, do we want to keep retrying?

Reply:
Good point, we might want to only catch the fetch failed error?

Reply:
Ok after talking offline we decided to keep this since it will likely spin anyways after adding auto-restart of the service... in most cases this will fail due to network being disconnected so this is probably still fine